### ISSUE TITLE
When Cookie/header size is large ~ more than 1200 bytes returns 408 Request Timeout Error 

### ISSUE THREAD

**Issue by venkatsarali** [haproxy/haproxy#64]
## Output of `haproxy -vv` and `uname -a`

```
HA-Proxy version 1.9.2 2019/01/16 - https://haproxy.org/
Build options :
  TARGET  = linux26
  CPU     = generic
  CC      = gcc
  CFLAGS  = -O2 -g -fno-strict-aliasing -Wdeclaration-after-statement -fwrapv -Wno-format-truncation -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-old-style-declaration -Wno-ignored-qualifiers -Wno-clobbered -Wno-missing-field-initializers -Wno-implicit-fallthrough -Wno-stringop-overflow -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_ZLIB=1 USE_REGPARM=1 USE_OPENSSL=1 USE_PCRE=1

Default settings :
  maxconn = 2000, bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with OpenSSL version : OpenSSL 1.0.2k-fips  26 Jan 2017
Running on OpenSSL version : OpenSSL 1.0.2k-fips  26 Jan 2017
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : SSLv3 TLSv1.0 TLSv1.1 TLSv1.2
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with zlib version : 1.2.8
Running on zlib version : 1.2.8
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with PCRE version : 8.21 2011-12-12
Running on PCRE version : 8.21 2011-12-12
PCRE library supports JIT : no (USE_PCRE_JIT not set)
Encrypted password support via crypt(3): yes
Built without multi-threading support (USE_THREAD not set).

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTX        side=FE|BE
              h2 : mode=HTTP       side=FE
       <default> : mode=HTX        side=FE|BE
       <default> : mode=TCP|HTTP   side=FE|BE

Available filters :
        [SPOE] spoe
        [COMP] compression
        [CACHE] cache
        [TRACE] trace

Linux test1 4.14.77-70.59.amzn1.x86_64 #1 SMP Mon Nov 12 22:02:45 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

## What's the configuration?

```
global
        log 127.0.0.1   local0 info
        maxconn 20000
        tune.ssl.default-dh-param 2048
        stats socket /tmp/haproxy
        ssl-default-bind-options no-sslv3
        ssl-default-bind-options no-tlsv10
        ssl-default-bind-ciphers ECDH+AESGCM:ECDH+CHACHA20:ECDH+AES256:ECDH+AES128:!aNULL:!SHA1

defaults
        log     global
        option httplog
        mode    http
        option  dontlognull
        retries 2
        option redispatch
        timeout connect 5000
        timeout client 50000
        timeout server 60000
        timeout http-request 30000
        timeout tarpit 30000
        timeout tunnel 86400s
        option abortonclose
        option allbackups
	  
backend client_cluster
        mode http
        cookie CLIENTINT insert indirect
        balance leastconn
        option forwardfor
        timeout server 300000
        option httpclose
        option http-use-htx
        redirect scheme https if !{ ssl_fc }
        server exampleclient1 l-exampleclient1:8080 cookie exampleclient1 check alpn http/1.1

frontend https-in
        bind *:443 ssl crt /etc/ssl/certs/sub.example.com.pem alpn http/1.1,h2
        mode http
        option http-use-htx
        reqadd X-Forwarded-Proto:\ https
        capture request header User-Agent len 200
        capture request header Referer len 150
        acl client_host hdr_sub(Host) -i sub.example.com
        use_backend client_cluster if client_host

frontend http-in
        bind 0.0.0.0:80
        mode http
        option http-use-htx
        capture request header User-Agent len 200
        capture request header Referer len 150
        acl client_host hdr_sub(Host) -i sub.example.com
        use_backend client_cluster if client_host
  
```

## Steps to reproduce the behavior

1. Set long length cookie in server side program (like login and pass session to another page)
2. When user login, server will send the set cookie to browser and when navigate to another page, Haproxy does not pass cookie back to server and getting 408 Request Timeout Error.
3. This issue happens only when we enable option http-use-htx and allow only http/1.1 to the backend server.
4. Tried following fine-tuning, but no use.   
        tune.maxrewrite 32768
        tune.bufsize 16384
        tune.bufsize 32768
        tune.h2.header-table-size 65535
        tune.maxaccept -1

## Do you have any idea what may have caused this?
Looks like HaProxy with enable option http-use-htx truncates cookie data when passing to backend server.

**Comment by wtarreau** [comment]
I could actually make your config fail under H2 due to the abortonclose option which we've found to have some issues. The good news is that I've tried again with Christopher's development fixes and that it fixes it. I didn't face any issue using a cookie 2.9kB long (even with abortonclose sent in HTTP/1 or HTTPS). I hope we'll merge the fixes early next week and will backport them in 1.9 so that you can try again. In the mean time, removing this option should very likely address the issue for you.

**Comment by wtarreau** [comment]
Fixed by the EOI patches backported to 1.9.6 and above.
